src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -598,6 +598,40 @@ static smt_termt convert_expr_to_smt(
     return convert_multiary_operator_to_terms(
       plus, converted, smt_bit_vector_theoryt::add);
   }
+  else if(std::any_of(
```

Comment:
:question: Why not just check if the result type of the expression - `plus.type()` is a pointer type?

Reply:
:pick: Also this looks like a relatively large block of code to add to an existing function. Maybe keep the if statement where it is, but move the body into a separate `convert_pointer_offset_addition` function?

Reply:
The first request has been addressed in b75e06cf7a0ee6497a63bac3da172313b5cd0963